src/tiled/mapeditor.cpp

```diff
@@ -1067,11 +1067,26 @@ EditableWangSet *MapEditor::currentWangSet() const
     return EditableWangSet::get(mWangDock->currentWangSet());
 }
 
+void MapEditor::setCurrentWangSet(EditableWangSet *wangSet)
+{
+    if (!wangSet) {
+        ScriptManager::instance().throwNullArgError(0);
+        return;
+    }
+    mWangDock->setCurrentWangSet(wangSet->wangSet());
+}
+
 int MapEditor::currentWangColorIndex() const
 {
     return mWangDock->currentWangColor();
 }
 
+void MapEditor::setCurrentWangColorIndex(int newIndex)
+{
```

Comment:
I noticed Tiled crashes if you call this when no terrain set is loaded / no map is open.

Tried adding this but it doesn't prevent it. Seems to be a crash trying to get the colorCount(), but currentWangSet() isn't null 

```cpp
    if(!mWangDock->currentWangSet())
        ScriptManager::instance().throwError(QCoreApplication::translate("Script Errors", "No wangset is loaded"));
```

Reply:
That check is fine, but you're forgetting to use `return;` to exit the function early. The `throwError` call is going to throw an error in the script engine, it's not a C++ exception.

Reply:
Thanks, that fixed it.